src/foam/dao/FUIDDAO.js

```diff
@@ -119,6 +119,12 @@ foam.CLASS({
       javaCode: `
         if ( getPropertyInfo().isDefaultValue(obj) ) {
           getPropertyInfo().set(obj, getUIDGenerator().getNext(null));
+        } else if ( getUIDGenerator() instanceof NUIDGenerator ) {
+          // NOTE: On replay, medusa node "put" its entries (with non-zero ids)
```

Comment:
This is part of replay in general, not specific to Medusa.

Reply:
JDAO replay doesn't call put() so didn't run through this block.